src/topology/metric_space/basic.lean

```diff
@@ -371,6 +371,16 @@ def sphere (x : α) (ε : ℝ) := {y | dist y x = ε}
 
 @[simp] theorem mem_sphere : y ∈ sphere x ε ↔ dist y x = ε := iff.rfl
 
+theorem sphere_is_empty_of_subsingleton [subsingleton α] (hε : ε ≠ 0) :
+  is_empty (sphere x ε) :=
```

Comment:
I think that we should have a version that says `sphere x r = 0` first.

Reply:
Done!